src/conduits/RwaInputConduit3.sol

```diff
@@ -0,0 +1,220 @@
+        psm = PsmAbstract(_psm);
+        dai = DSTokenAbstract(PsmAbstract(_psm).dai());
+        gem = _gem;
+        to = _to;
+        quitTo = _quitTo;
+
+        // Give unlimited approve to PSM gemjoin
+        _gem.approve(address(PsmAbstract(_psm).gemJoin()), type(uint256).max);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /*//////////////////////////////////
+               Authorization
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /*//////////////////////////////////
+               Administration
+    //////////////////////////////////*/
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"quitTo"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
+            require(data != address(0), "RwaInputConduit3/invalid-quit-to-address");
+            quitTo = data;
+        } else {
+            revert("RwaInputConduit3/unrecognised-param");
+        }
+
+        emit File(what, data);
+    }
+
+    /*//////////////////////////////////
+               Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Method to swap USDC contract balance to DAI through PSM and push it into RwaUrn address.
+     * @dev `msg.sender` must first receive push acess through mate().
+     */
+    function push() external {
+        require(may[msg.sender] == 1, "RwaInputConduit3/not-mate");
+        uint256 balance = gem.balanceOf(address(this));
+        require(balance > 0, "RwaInputConduit3/insufficient-gem-balance");
+
+        psm.sellGem(address(this), balance);
+
+        uint256 daiBalance = dai.balanceOf(address(this));
+        dai.transfer(to, daiBalance);
+
+        emit Push(to, daiBalance);
+    }
+
+    /**
+     * @notice Flushes out any GEM balance to `quitTo` address.
+     * @dev Can only be called by an admin.
+     */
+    function quit() external auth {
```

Comment:
Why is this function `auth`ed, any reason we don't let the `operator` call this?

Reply:
you are right, operator the one who should call this

Reply:
We don't have operators in input conduits, only admins and mates.

Reply:
I did it callable by `mate`

Reply:
Alright, I was just worried about the nomenclature we use here, because it can become confusing 😅. All good.